plugin-server/src/cdp/hog-executor.ts

```diff
@@ -327,15 +327,51 @@ export class HogExecutor {
         }
     }
 
-    execute(invocation: HogFunctionInvocation): HogFunctionInvocationResult {
+    executeTransformation(invocation: HogFunctionInvocation): HogFunctionInvocationResult {
+        const result = this.execute(invocation, {
+            allowAsync: false,
+            requireResult: true,
+            functionType: 'transformation',
+        })
+
+        // For transformations, we require a result to be returned
+        if (result.finished && result.execResult === undefined) {
+            result.error = 'Transformation function must return a value'
+            result.logs.push({
+                level: 'error',
+                timestamp: DateTime.now(),
+                message: 'Transformation function must return a value',
+            })
+        }
+
+        return result
+    }
+
+    executeDestination(invocation: HogFunctionInvocation): HogFunctionInvocationResult {
+        return this.execute(invocation, {
```

Comment:
I think this might be a premature refactor. Reason being, once you start implementing the "transformation" code I think you'll see what I mean which is that we will want to _pass in_ functions to the execution. Destinations will pass in async_functions like `fetch` and transformations will have normal functions like `geoipLocate` etc.

My gut tells me that we would want the `execute` function to not have a concept of these options but rather simply receive a bunch of functions that it can run instead. 

this would also make running unit tests easier as we would be injecting the functions into the exec command from the top level. Not sure if any of this makes sense (my brain is already thinking ahead to the actual use case for this stuff)